synapse/storage/databases/main/devices.py

```diff
@@ -139,6 +139,27 @@ async def get_devices_by_user(self, user_id: str) -> Dict[str, Dict[str, str]]:
 
         return {d["device_id"]: d for d in devices}
 
+    async def get_devices_by_auth_provider_session_id(
+        self, auth_provider_id: str, auth_provider_session_id: str
+    ) -> List[Dict[str, Any]]:
+        """Retrieve the list of devices associated with an OIDC session ID (sid).
```

Comment:
```suggestion
        """Retrieve the list of devices associated with a SSO IdP session ID.
```

(This method seems unused here, but I think it's use is fairly obvious if we're saving this into the database...)

Reply:
Yep, and definitely used in the OIDC logout PR afterwards